packages/plugin-ext/src/main/browser/scm-main.ts

```diff
@@ -222,13 +224,15 @@ export class PluginScmProvider implements ScmProvider {
                 const { start, deleteCount, rawResources } = groupSlice;
                 const resources = rawResources.map(rawResource => {
                     const { handle, sourceUri, icons, tooltip, strikeThrough, faded, contextValue, command } = rawResource;
-                    const icon = icons[0];
-                    const iconDark = icons[1] || icon;
+
+                    const icon = ThemeIcon.isThemeIcon(icons[0]) ? ThemeIcon.asClassName(icons[0]) : this.toUrl(vscodeURI.revive(icons[0]));
+                    const iconDark = ThemeIcon.isThemeIcon(icons[1]) ? ThemeIcon.asClassName(icons[1]) : this.toUrl(vscodeURI.revive(icons[1])) || icon;
+
                     // eslint-disable-next-line @typescript-eslint/no-explicit-any
                     const colorVariable = (rawResource as any).colorId && this.colors.toCssVariableName((rawResource as any).colorId);
                     const decorations = {
-                        icon: icon ? vscodeURI.revive(icon) : undefined,
-                        iconDark: iconDark ? vscodeURI.revive(iconDark) : undefined,
+                        icon: icon,
+                        iconDark: iconDark,
```

Comment:
minor: since the property name is the same as the variable we can actually do the following:

```suggestion
                        icon,
                        iconDark,
```